File: Clients/mc.market.reactwebapp (old)/src/features/listings/types/listing.ts
export type ListingSort = "newest" | "priceAsc" | "priceDesc";

export interface ListingParams {
  categoryId?: number;
  sort?: ListingSort;
}

export interface ListingParameter {
  name: string;
  value: string;
}

export interface Category {
  id: number;
  name: string;
}

export interface Listing {
  id: string;
  ownerId: string;
  title: string;
  description: string;
  categoryId: number;
  category: Category;
  createdAt: string;
  price: number;
  stock: number;
  images: string[];
  tags: string[];
  parameters: ListingParameter[];
  location?: {
    country?: string | null;
    region?: string | null;
    city?: string | null;
  };
  condition?: string;
  sellerRating?: number;
  sellerSales?: number;
}

export interface ListingView {
  id: string;
  title: string;
  description: string;
  price: number;
  imageId: string;
}

export interface ListingPaginatedResponse {
  items: ListingView[];
  pageSize: number;
  pageIndex: number;
  totalPages: number;
  hasNextPage: boolean;
  hasPreviousPage: boolean;
}

export function listingImageUrl(
  photoId: string,
  width: number,
  height: number,
) {
  return `https://images.unsplash.com/${photoId}?w=${width}&h=${height}&fit=crop&auto=format`;
}

export function listingSku(id: string | number) {
  return `MKT-${String(id).padStart(4, "0")}`;
}

export function sellerInitials(name: string) {
  return name
    .replaceAll(".", "")
    .split(" ")
    .filter(Boolean)
    .map((part) => part[0])
    .join("")
    .slice(0, 2)
    .toUpperCase();
}

File: Services/Catalog/MC.Catalog.Domain/Entities/Listing.cs
namespace MC.Catalog.Domain.Entities;

public class Listing
{
    public string Id { get; set; } = string.Empty;
    public Guid OwnerGuid { get; set; } = Guid.Empty;
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public uint CategoryId { get; set; }
    public Category Category { get; set; } = new();
    public uint LocationId { get; set; }
    public Location Location { get; set; } = new();
    public DateTimeOffset CreatedAt { get; set; }
    public decimal Price { get; set; }
    public int StockQuantity { get; set; }
    public List<string> ImageLinks { get; set; } = [];
    public List<string> Tags { get; set; } = [];
    public List<Tuple<string, string>> Parameters { get; set; } = [];
}
